runtime/expr.h

```diff
@@ -533,6 +533,18 @@ inline bool is_true(const expr& x) {
   return cx ? cx->value != 0 : false;
 }
 SLINKY_ALWAYS_INLINE inline bool is_false(const expr& x) { return is_zero(x); }
+SLINKY_ALWAYS_INLINE inline bool is_logical(const expr& x) {
```

Comment:
At the risk of premature optimization, we could rewrite this to be a single bit-test operation instead of a switch (eg `constexpr mask = (1 << expr_node_type::less) | ...; return ((1 << x.type()) & mask) != 0;`)